core/common/src/main/java/alluxio/collections/IndexedSet.java

```diff
@@ -128,21 +139,82 @@
   }
 
   /**
+   * An interface extending {@link FieldIndex}, represents a unique index. A unique index is an
+   * index where each index value only maps to one object.
+   *
+   * @param <T> type of objects in this {@link IndexedSet}
+   */
+  public interface UniqueFieldIndex<T> extends FieldIndex<T> {
```

Comment:
BTW, if we implement my suggestion that makes FieldIndex manage the representation of its values, it will no longer be necessary to make the interface private to avoid incorrect uses